### PR TITLE
Component config: Improve and fix code style, hidden field + Add style for fieldset description

### PR DESCRIPTION
**Pull request by cyrez**
#### Summary of Changes
- Fix code style
- Fix hidden field (no html to be rendered when field is hidden, to prevent empty lines (due to css classes for fields)
- Add style alert-info + icon for the description of a fieldset (description in top of each tab, when filled)
#### Testing Instructions

**Test for fieldset description style:**
- Go to any component configuration page (Options) and check description has now an alert-info look

**Before Patch**:
capture d ecran 2016-05-24 a 01 07 00

**After Patch**:
capture d ecran 2016-05-24 a 01 07 30

**Test for hidden fields fix:**
- To test fix for hidden fields, add in a config.xml those lines : 

```
        <field
            name="hidden1"
            type="hidden"
            default="1"
        />

        <field
            name="hidden2"
            type="hidden"
            default="1"
        />

        <field
            name="hidden3"
            type="hidden"
            default="1"
        />

        <field
            name="hidden4"
            type="hidden"
            default="1"
        />
```

**Before Patch**:
capture d ecran 2016-05-25 a 12 47 05

**After Patch**:
capture d ecran 2016-05-25 a 12 47 37
- You can control after save, that hidden fields are saved in the database (#__extensions > component params)

### CODE REVIEW

**Comment by andrepereiradasilva** [comment]
Actually, not related to your Pr, but i think that message is not correct.
I mean the category can also override article settings.

File: administrator/components/com_config/view/component/tmpl/default.php [hunk 1]
```diff
@@ -37,14 +37,16 @@
 
 <form action="<?php echo JRoute::_('index.php?option=com_config'); ?>" id="component-form" method="post" name="adminForm" autocomplete="off" class="form-validate form-horizontal">
 	<div class="row-fluid">
+
 		<!-- Begin Sidebar -->
-		<div id="sidebar" class="span2">
+		<div class="span2" id="sidebar">
```
Comment (andrepereiradasilva) [inline administrator/components/com_config/view/component/tmpl/default.php]: why did you change this? IMHO is better to have `id`, `name` and `type` attributes (when they exist) first.

File: administrator/components/com_config/view/component/tmpl/default.php [hunk 2]
```diff
@@ -37,14 +37,16 @@
...
 			<div class="sidebar-nav">
 				<?php echo $this->loadTemplate('navigation'); ?>
 			</div>
-		</div>
-		<!-- End Sidebar -->
-		<div class="span10">
+		</div><!-- End Sidebar -->
+
+		<div class="span10" id="config">
+
```
Comment (andrepereiradasilva) [inline administrator/components/com_config/view/component/tmpl/default.php]: same as above

File: administrator/components/com_config/view/component/tmpl/default.php [hunk 3]
```diff
@@ -63,55 +65,55 @@
...
 					<?php endif; ?>
 					<?php $label = empty($fieldSet->label) ? 'COM_CONFIG_' . $name . '_FIELDSET_LABEL' : $fieldSet->label; ?>
-					<li<?php echo $rel; ?>><a href="#<?php echo $name; ?>" data-toggle="tab"><?php echo JText::_($label); ?></a></li>
+					<li<?php echo $rel; ?>><a data-toggle="tab" href="#<?php echo $name; ?>"><?php echo JText::_($label); ?></a></li>
 				<?php endforeach; ?>
-			</ul>
-			<div class="tab-content">
+			</ul><!-- /configTabs -->
+
+			<div class="tab-content" id="configContent">
```
Comment (andrepereiradasilva) [inline administrator/components/com_config/view/component/tmpl/default.php]: same as above

File: administrator/components/com_config/view/component/tmpl/default.php [hunk 4]
```diff
@@ -63,55 +65,55 @@
 						<?php $rel = ' data-showon=\'' . json_encode($showonarr) . '\''; ?>
 					<?php endif; ?>
 					<?php $label = empty($fieldSet->label) ? 'COM_CONFIG_' . $name . '_FIELDSET_LABEL' : $fieldSet->label; ?>
-					<li<?php echo $rel; ?>><a href="#<?php echo $name; ?>" data-toggle="tab"><?php echo JText::_($label); ?></a></li>
+					<li<?php echo $rel; ?>><a data-toggle="tab" href="#<?php echo $name; ?>"><?php echo JText::_($label); ?></a></li>
```
Comment (andrepereiradasilva) [inline administrator/components/com_config/view/component/tmpl/default.php]: same as above, href is the more important field in a anchor, so IMHO it should go first

File: administrator/components/com_config/view/component/tmpl/default.php (hunk 1 above)
Comment (JoomliC) [inline administrator/components/com_config/view/component/tmpl/default.php]: Code standards HTML : class, id, data-*, Everything else (https://joomla.github.io/coding-standards/?coding-standards/chapters/html.md)
IMO, BS 3 demo site use this order too ;-)

File: administrator/components/com_config/view/component/tmpl/default.php (hunk 2 above)
Comment (JoomliC) [inline administrator/components/com_config/view/component/tmpl/default.php]: Same as above ;-)

File: administrator/components/com_config/view/component/tmpl/default.php (hunk 1 above)
Comment (andrepereiradasilva) [inline administrator/components/com_config/view/component/tmpl/default.php]: ok then! my bad

File: administrator/components/com_config/view/component/tmpl/default.php (hunk 4 above)
Comment (JoomliC) [inline administrator/components/com_config/view/component/tmpl/default.php]: Same as above: code standards ;-)

File: administrator/components/com_config/view/component/tmpl/default.php [hunk 5]
```diff
@@ -63,55 +65,55 @@
...
 				<?php endforeach; ?>
-			</div>
-		</div>
-	</div>
-	<div>
+			</div><!-- /configContent -->
+
+		</div><!-- /config -->
+
 		<input type="hidden" name="id" value="<?php echo $this->component->id; ?>" />
```
Comment (andrepereiradasilva) [inline administrator/components/com_config/view/component/tmpl/default.php]: why are this input idented?

**Comment by JoomliC** [comment]
> Actually, not related to your Pr, but i think that message is not correct.
I mean the category can also override article settings.

Exactly: nothing related to this PR, as screenshot is just an example ;-)

File: administrator/components/com_config/view/component/tmpl/default.php (hunk 5 above)
Comment (JoomliC) [inline administrator/components/com_config/view/component/tmpl/default.php]: I would say: read the code standards for HTML, and then, be back ;-)
This not required, but sometimes could help to identify close block.

File: administrator/components/com_config/view/component/tmpl/default.php (hunk 2 above)
Comment (andrepereiradasilva) [inline administrator/components/com_config/view/component/tmpl/default.php]: ok then! my bad

File: administrator/components/com_config/view/component/tmpl/default.php (hunk 4 above)
Comment (andrepereiradasilva) [inline administrator/components/com_config/view/component/tmpl/default.php]: ok then! my bad

**Comment by andrepereiradasilva** [comment]
you're right. i should have read the HTML coding standard before. It seems all ok.

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on 497e7dbb626db160ef146f65ffb78e4ea8da45da<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10622">issues.joomla.org/joomla-cms/10622</a>.</sub>

**Comment by JoomliC** [comment]
@andrepereiradasilva Thanks for testing!

> you're right. i should have read the HTML coding standard before. It seems all ok.

Not exactly what i meant, but i asked to be sure you understand it as i did for HTML code standards, and so that we agree! ;-)

**Comment by joomlara** [comment]
I have tested this item :white_check_mark: successfully on 497e7dbb626db160ef146f65ffb78e4ea8da45da<br /><br />Tested successfully<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10622">issues.joomla.org/joomla-cms/10622</a>.</sub>

**Comment by joomlara** [comment]
Tested successfully.  <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10622">issues.joomla.org/joomla-cms/10622</a>.</sub>

**Comment by zero-24** [comment]
RTC based on testing<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10622">issues.joomla.org/joomla-cms/10622</a>.</sub>

**Comment by roland-d** [comment]
Thanks everybody<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10622">issues.joomla.org/joomla-cms/10622</a>.</sub>
